[WIP] General cleanup of network/router docs.

pecameron opened the pull request:
[WORK IN PROGRESS - DONOT MERGE]
At this point I would like to get a discussion going on 
what to do about the networking docs. I find it difficult to 
find what I am looking for and I expect others do as well.
Also, we don't have complete and accurate information on
some of the topics. 
 
Toward that end it is helpful to clone and run asciibinder
and use the browser to go through this.

Doc team: Is this going in a reasonable direction? Are there
additional things that need to be done, or done differently?

There are some rough notes included that will be cleaned up over time.
Feel free to comment on those as well.

Review and comments are always appreciated.

I think of the docs for a new user to start from the top and get an overview. Toward that end
my changes start with architecture/index.adoc bottom section.

pecameron commented:
@openshift/team-documentation @openshift/networking  
Starting to make changes to the network docs. I am hoping for a lot of feedback to help change this into what we want our customers to use.

danwinship commented:
> Are there additional things that need to be done, or done differently?

Note that #3575 (which will merge soon) splits all the networking-related stuff from admin_guide/managing_pods.adoc out into a new admin_guide/managing_networking.adoc, so that might also be a good destination for some of this stuff. (I haven't looked at your changes in detail yet.)

vikram-redhat commented:
Thanks @pecameron. @bfallonf , @aheslin  FYI.

pecameron commented:
Just did a rebase...

pecameron commented:
Put Nuage SDN discussion in a separate file.

pecameron commented:
reorganized externalIP, ingressIP

pecameron commented:
Another rebase and some more edits.

vikram-redhat commented:
@mburke5678 FYI. It maybe a good idea for @pecameron and you to have a meeting to discuss how you can help out with regards to https://bugzilla.redhat.com/show_bug.cgi?id=1421924.

@pecameron - Michael has a very specific focus here on that particular bug. If I understand correctly, your PR is an overall look at the networking docs. 

CC: @aheslin

pecameron commented:
@vikram-redhat I am starting at the top and working my way to the details. This pass is just to make things a little easier to find and use stuff. I expect there will be a lot more to do after this. 

@mburke5678 I am in Westford, 2w655. Drop by some time and we can talk.

mburke5678 commented:
Thanks!

On Wed, Mar 22, 2017 at 12:49 PM, Phil Cameron <notifications@github.com>
wrote:

> @vikram-redhat <https://github.com/vikram-redhat> I am starting at the
> top and working my way to the details. This pass is just to make things a
> little easier to find and use stuff. I expect there will be a lot more to
> do after this.
>
> @mburke5678 <https://github.com/mburke5678> I am in Westford, 2w655. Drop
> by some time and we can talk.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/3943#issuecomment-288463589>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYCpQxzmsOuvMuNBcl_51zUvUSNDgyk5ks5roVETgaJpZM4MeHhW>
> .
>



-- 
Technical Writer, Customer Content Services
Customer Experience & Engagement (CEE)
Red Hat, Inc.

pecameron commented:
Put F5 in architecture into separate file. More general edits.

pecameron commented:
@vikram-redhat @bfallonf @knobunc  I would like to wrap up this round of changes. There is a lot more that can be done but I would like to get some of this merged to be a basis of future changes. Would you have a chance to go through it from an organization and content view. We can do the detail editing pass next. Thanks!

bfallonf commented:
@pecameron  I've been hesitant to comment, as every day I come in there's more changes, but if you think this is in a state to look at, sure.

You've actually started to put efforts into something I've been looking at lately, which is the Architecture guide, and how right now it's quite messy, and there's a lot of "procedure stuff" when I think a lot of that should be "context and reference stuff". But you're trying to organise it more, not specifically target that, which is good.

Overall, I think at the minimum, the architecture/routes file served better as a house for the contextual stuff, and dividing it up into different files might actually make it harder to navigate and go against what the architecture guide is all about. And I'm thinking the same about the SDN stuff. It's the same premise, spread over a few files talking about the different methods. I think that would be better as one file with different sections, each talking about the referential stuff, leaving the Install/Admin Guides to talk about the procedures.

So this is my way of saying: You've done great in moving the right information into the right place, but I think dividing what is essentially two concepts into a bunch of files not not be the best approach. I've written up a quick suggestion google doc with what I'd imagine this could be. It might not be what you are thinking of first, but I think if what information is currently there is condensed, and after your suggestions (to the above, and what you've done with this PR) it could be the best of both worlds?

https://docs.google.com/a/redhat.com/document/d/1erR5hVvI3A1j6hwRU2LM_Ca0NeP16adBHcYi4YJJrcw/edit?usp=sharing

Let me know what you think.

pecameron commented:
@bfallonf I need to stop and take a breath. Everywhere I look there is more to write. So I would like to stop here and figure out what to do with the changes in this PR. I went too far and got carried away.

As you point out, there are some fundamental decisions that need to be made and then we need to work to bring them to life. I read your google doc and added comments on my thinking about this. Bit-rot happens fast in the few doc files that we have and I am not sure whether to continually resolve them and keep this PR up to date or just let the PR rest.

We have a different approach to documenting this and I am not sure how we should proceed.

bfallonf commented:
@pecameron You're right that the further this goes the more complicated it gets. It's a viscous cycle of git conflicts here (there are more there right now).

How about this:

Are your efforts here concerning a specific 3.5 endeavor? Is there new content that's been added here that is necessary to be included in the 3.5 docs? Is there a trello card somewhere that needs to be moved to the "completed" column? (I ask because you mention a few comments above that you want to "wrap up this round of changes"). Like I said, I was planning on having a deep look at the Architecture Guide after the 3.5 release. I'm wondering if you'd be happy with merging any necessary stuff now (either in this or a separate PR), and the two of us work together to organise the routing/networking docs in the next few weeks? I don't want my own thoughts on what's needed to get in the way of your deliverables, and if we have more time perhaps we'll come to better, more agreeable conclusions. WDYT?

cc: @vikram-redhat @aheslin

pecameron commented:
@bfallonf 
Sorry about the delay in this response. I missed your comment (my bad).
This is not specific to 3.5 It was a general attempt to make it easier for me to find stuff in the docs that came out of my frustration trying to find what I need and stumbling onto answers long after I gave up on finding them. I now just grep the .adoc files for what I need. The information is mostly there its a matter of finding it.

We added a trello card, https://trello.com/c/jRTbijXN/456-5-doc-cleanup, with a vague placeholder to make changes. It is/was pretty open ended. It just tracks me working on docs. This is a large topic and it will not be done quickly or in a single pass. I expect changes will continue to be made over time. So we will finish this card and possible start others. 

I started at the top of the architecture section and just got carried away. I ended up morphing this into something that I can use (instead of grep) but I am not at all sure others would find it useful. There are lots of ways of doing this and I am open to different approaches.

I would love to work with you on this material. Whatever seems interesting can go in whenever the timing is right. I don't think this is an all or nothing set of changes. It evolved by trying to find out about something and adding cross-references from the various places the information shows up. I also moved some discussions to different sections where they seemed to belong. 

 cc: @vikram-redhat @ahardin-rh @knobunc

bfallonf commented:
@pecameron No problem.

And please rest assured that I feel your pain. I've organised to have a meeting post-3.5 release with Ali, our content strategist, to talk about what I want out of the Architecture Guide. I'll make sure to include some of your ideas above, because I think there's some good stuff there. I think your findability concerns are real.

I'll keep you updated!

bfallonf commented:
@pecameron Right, sorry, this has been on my to-do list, but not forgotten. I have a plan. I'll try get your changes into the docs in the next couple days, but attempt the Architecture changes in a different PR. I'll keep this open for now, just in case there's any opposition.

gaurav-nelson commented:
Thanks @bfallonf 

👋
